### DISCUSSION TITLE
Is it possible to have multiple calls for packer?

### DISCUSSION THREAD

**Discussion by samrathchadha** [wbthomason/packer.nvim#536] [Q&A] [answered]
Is there a way I can call the `require("packer").startup` many times? Or is there any equivalent?

**Comment by wbthomason** [comment]
You can take the more manual approach of calling `packer.init()` once yourself, then calling `use` as many times as you need. You can take a look at my dotfiles (https://github.com/wbthomason/dotfiles/blob/linux/neovim/.config/nvim/lua/plugins.lua) for an example of this style.

**Reply by samrathchadha** [reply]
I'll try it out in a bit. Thanks for the response

**Comment by grundic** [comment, marked as answer]
I found myself in the same situation and it seems I found nice solution for this.

Some background about my setup: I'm using `vcsh` and `mr` to manage my dotfiles. Any of the tool configuration could have multiple sources: the primary one, which I use everywhere and keep in my public repository and auxiliary one, which is usually referenced to particular environment (e.g. corporate settings) and is stored secretly.

Therefore, I have my primary `plugins.lua` with the following content:

**plugins.lua**
```lua

return require('packer').startup(function()
  -- Packer can manage itself                                                                                                                                                                   
  use 'wbthomason/packer.nvim'   

  ...
  ...


  -- Dynamically install other plugins from custom sources.
  vim.cmd('runtime! lua/extra-plugins*.lua')
end)
```

And I can have extra plugins installed like this:

**extra-plugins-corporate.lua**
```lua
  local packer = require('packer')
  packer.use 'additional-plugin'
```

It took me quite a while to figure out how to dynamically include other plugins to the Packer. Probably it's worth adding to documentation to make it clear.
